### PR TITLE
Add gulp task to run unit tests

### PR DESCRIPTION
**Pull request by ChuckFerring**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -49,6 +50,11 @@ gulp.task('raml-lint', () => {
         .pipe(eslint.failAfterError());
 });
 
-gulp.task('ci-checks', ['js-lint', 'json-lint', 'xml-lint', 'validate-naming']);
+gulp.task('unittest', function () {
+  return gulp.src(['**/js/tests/unitTests.js', NO_MODULES])
+    .pipe(ava({verbose: true}))
+});
+
+gulp.task('ci-checks', ['js-lint', 'json-lint', 'xml-lint', 'validate-naming', 'unittest']);
```
Comment (shalinijoshi19) [inline gulpfile.js]: >js-lint', 'json-l [](start = 25, length = 17)

Do you know if js-lint will check for spacing issues, style issues (eg enclosing if blocks in braces etc, position of braces like the ts-linter we use for opent2t repo?

File: org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ function findResource(schema, di, resourceId) {
         return d.di === di; 
     }); 
     
-    if (!entity) throw new Error('Entity - '+ di +' not found.');
+    if (!entity) throw new Error('NotFound');
```
Comment (shalinijoshi19) [inline org.opent2t.sample.thermostat.superpopular/com.nest.thermostat/js/thingTranslator.js]: > new Er [](start = 22, length = 7)

nit: braces for if blocks even if one-line statements

File: verifiers/package.json [hunk 3]
```diff
@@ -10,12 +10,14 @@
...
+    "gulp-ava": "^0.16.0",
     "gulp-eslint": "^3.0.1",
     "gulp-util": "^3.0.7",
     "gulp-xml-validator": "^0.1.2",
     "raml-1-parser": "^1.1.8",
     "through2": "^2.0.1",
     "xmldom": "^0.1.22",
-    "xpath": "0.0.23"
+    "xpath": "0.0.23",
+    "opent2t-test-mocks": "file:../test"
```
Comment (shalinijoshi19) [inline verifiers/package.json]: >file [](start = 27, length = 4)

neat thanks

**Comment by shalinijoshi19** [comment]
:shipit:

**Comment by danielrischar** [comment]
:shipit:

**Pull request merged**